jwk/helper.go

```diff
@@ -120,6 +128,27 @@ func FindKeyByPrefix(set *jose.JSONWebKeySet, prefix string) (key *jose.JSONWebK
 	return First(keys.Keys), nil
 }
 
+func FindPublicKey(set *jose.JSONWebKeySet) (key *jose.JSONWebKey, err error) {
+	keys := ExcludeKeysByPrefix(set, "private")
+	if len(keys.Keys) == 0 {
+		return nil, errors.New("key not found")
+	}
+
+	return First(keys.Keys), nil
+}
+
+func ExcludeKeysByPrefix(set *jose.JSONWebKeySet, prefix string) *jose.JSONWebKeySet {
+	keys := new(jose.JSONWebKeySet)
+
+	for _, k := range set.Keys {
+		if !strings.HasPrefix(k.KeyID, prefix+":") {
```

Comment:
We can actually use type assertion here on jose.JSONWebKey to check if the thing is a private key or not (like here https://github.com/ory/hydra/pull/2459/files#diff-d2d06a70d7e57e038f22beffe343a39d7efa66e3be691e00342697c5b7a909c9R125 ). I think that would be even better than doing this prefix stuff here.

Reply:
@aeneasr : will do the changes